File: Proptimo/Backend/Proptimo/Core/Proptimo.Domain/Entities/RealEstateTypeFeature.cs
using Proptimo.Domain.Entities.Common;
using Proptimo.Domain.Entities.Enum;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Proptimo.Domain.Entities
{
    public class RealEstateTypeFeature : BaseEntity
    {
        public string Name { get; set; }
        public TypeFeatureDataType DataType { get; set; }
        public bool IsUnit { get; set; }
        public bool IsRequired { get; set; }
        public List<string>? Options { get; set; } 

        public string RealEstateTypeId { get; set; }
        public RealEstateType RealEstateType { get; set; }
    }
}

File: Proptimo/Frontend/proptimo/src/features/api/types/realEstateTypeFeature.ts
export const TypeFeatureDataType = {
    Int: 0,
    Decimal: 1,
    Bool: 2,
    String: 3,
    DateTime: 4
} as const;

export type TypeFeatureDataType = typeof TypeFeatureDataType[keyof typeof TypeFeatureDataType];

export interface RealEstateTypeFeature {
    id: string,
    name: string,
    dataType : TypeFeatureDataType,
    isUnit : boolean,
    isRequired: boolean,
    options: string[]
    realEstateTypeId: string
}

export interface CreateRealEstateTypeFeature{
    name: string,
    dataType : TypeFeatureDataType,
    isUnit : boolean,
    isRequired: boolean,
    options: string[]
    realEstateTypeId: string
}

export interface UpdateRealEstateTypeFeature {
    id: string,
    name: string,
    dataType : TypeFeatureDataType,
    isUnit : boolean,
    isRequired: boolean,
    options: string[]
    realEstateTypeId: string
}

export interface GetByIdRealEstateTypeFeature {
    id: string,
    name: string,
    dataType : TypeFeatureDataType,
    isUnit : boolean,
    isRequired: boolean,
    options: string[]
    realEstateTypeId: string
}

export interface GetAllEstateTypeFeaturesByTypeId{
    data: RealEstateTypeFeature[];
    total: number;
}



